react-components/src/components/GiftCardOptions/giftCardOptions.js

```diff
@@ -0,0 +1,376 @@
+/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ ~ Copyright 2021 Adobe
+ ~
+ ~ Licensed under the Apache License, Version 2.0 (the "License");
+ ~ you may not use this file except in compliance with the License.
+ ~ You may obtain a copy of the License at
+ ~
+ ~     http://www.apache.org/licenses/LICENSE-2.0
+ ~
+ ~ Unless required by applicable law or agreed to in writing, software
+ ~ distributed under the License is distributed on an "AS IS" BASIS,
+ ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ ~ See the License for the specific language governing permissions and
+ ~ limitations under the License.
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+import React, { useEffect, useState } from 'react';
+import { defineMessages, useIntl } from 'react-intl';
+
+import LoadingIndicator from '../LoadingIndicator';
+import { useAwaitQuery } from '../../utils/hooks';
+import GIFT_CARD_PRODUCT_QUERY from '../../queries/query_gift_card_product.graphql';
+import { useConfigContext } from '../../context/ConfigContext';
+import Price from '../Price';
+
+const OPEN_AMOUNT = 'open_amount';
+
+const messages = defineMessages({
+    amountSelectionLabel: {
+        id: 'product:gift-card-amount-selection',
+        defaultMessage: 'Amount'
+    },
+    chooseAmount: {
+        id: 'product:gift-card-choose-amount',
+        defaultMessage: 'Choose an Amount'
+    },
+    otherAmount: {
+        id: 'product:gift-card-other-amount',
+        defaultMessage: 'Other Amount'
+    },
+    customAmountLabel: {
+        id: 'product:gift-card-custom-amount',
+        defaultMessage: 'Amount in'
+    },
+    customPriceMinimum: {
+        id: 'product:gift-card-custom-ammount-minimum',
+        defaultMessage: 'Minimum'
+    },
+    customPriceMaximum: {
+        id: 'product:gift-card-custom-ammount-maximum',
+        defaultMessage: 'Maximum'
+    }
+});
+
+const GiftCartOptions = () => {
+    const [giftCardState, setGiftCardState] = useState(null);
+    const giftCardProductQuery = useAwaitQuery(GIFT_CARD_PRODUCT_QUERY);
+
+    const { formatMessage } = useIntl();
+    const {
```

Comment:
If you would use the `PortalPlacer` in Venia to place the component, it will take care of providing dataset values as React props and you don't need to do any DOM queries here. 

https://github.com/adobe/aem-core-cif-components/blob/master/react-components/src/components/PortalPlacer/PortalPlacer.js